File: 01jupyter/nb/source/11_simd/include/omp_simd.f90
subroutine saxpy(n, a, x, y)
  integer :: n
  real(8) :: a, x(n), y(n)
  integer :: i
  !$omp simd
  do i = 1, n
     y(i) = a * x(i) + y(i)
  end do
end subroutine saxpy

File: 01jupyter/nb/source/11_simd/problems/00_omp_simd/omp_simd.cpp
/* y[i] = a * x[i] + y[i] (saxpy/axpy) を n 要素について行う */
void saxpy(long n, double a, double * x, double * y) {
  // BEGIN ANSWER: 下の for ループの直前に #pragma omp simd を1行追加し, このループをSIMD化せよ.
  #pragma omp simd
  // END ANSWER
  for (long i = 0; i < n; i++) {
    y[i] = a * x[i] + y[i];
  }
}
